src/EtherFiRedemptionManager.sol

```diff
@@ -0,0 +1,280 @@
+        uint256 afterEEthAmount = eEth.balanceOf(address(this));
+
+        uint256 transferredEEthAmount = afterEEthAmount - beforeEEthAmount;
+        _redeem(transferredEEthAmount, receiver);
+    }
+
+    /**
+     * @notice Redeems weETH for ETH.
+     * @param weEthAmount The amount of weETH to redeem after the exit fee.
+     * @param receiver The address to receive the redeemed ETH.
+     * @param owner The address of the owner of the weETH.
+     */
+    function redeemWeEth(uint256 weEthAmount, address receiver, address owner) public whenNotPaused nonReentrant {
+        uint256 eEthShares = weEthAmount;
+        uint256 eEthAmount = liquidityPool.amountForShare(eEthShares);
+        require(weEthAmount <= weEth.balanceOf(owner), "EtherFiRedemptionManager: Insufficient balance");
+        require(canRedeem(eEthAmount), "EtherFiRedemptionManager: Exceeded total redeemable amount");
+
+        uint256 beforeEEthAmount = eEth.balanceOf(address(this));
+        IERC20(address(weEth)).safeTransferFrom(owner, address(this), weEthAmount);
+        weEth.unwrap(weEthAmount);
+        uint256 afterEEthAmount = eEth.balanceOf(address(this));
+
+        uint256 transferredEEthAmount = afterEEthAmount - beforeEEthAmount;
+        _redeem(transferredEEthAmount, receiver);
+    }
+
+
+    /**
+     * @notice Redeems ETH.
+     * @param ethAmount The amount of ETH to redeem after the exit fee.
+     * @param receiver The address to receive the redeemed ETH.
+     */
+    function _redeem(uint256 ethAmount, address receiver) internal {
+        _updateRateLimit(ethAmount);
+
+        uint256 ethShares = liquidityPool.sharesForAmount(ethAmount);
+        uint256 ethShareToReceiver = ethShares.mulDiv(BASIS_POINT_SCALE - exitFeeInBps, BASIS_POINT_SCALE);
+        uint256 eEthAmountToReceiver = liquidityPool.amountForShare(ethShareToReceiver);
+
+        uint256 prevLpBalance = address(liquidityPool).balance;
+        uint256 sharesToBurn = liquidityPool.sharesForWithdrawalAmount(eEthAmountToReceiver);
+
+        uint256 ethShareFee = ethShares - sharesToBurn;
+        uint256 feeShareToTreasury = ethShareFee.mulDiv(exitFeeSplitToTreasuryInBps, BASIS_POINT_SCALE);
+        uint256 eEthFeeAmountToTreasury = liquidityPool.amountForShare(feeShareToTreasury);
+        uint256 feeShareToStakers = ethShareFee - feeShareToTreasury;
+
+        // Withdraw ETH from the liquidity pool
+        uint256 prevBalance = address(this).balance;
+        assert (liquidityPool.withdraw(address(this), eEthAmountToReceiver) == sharesToBurn);
+        uint256 ethReceived = address(this).balance - prevBalance;
+
+        // To Stakers by burning shares
+        eEth.burnShares(address(this), feeShareToStakers);
+
+        // To Treasury by transferring eETH
+        IERC20(address(eEth)).safeTransfer(treasury, eEthFeeAmountToTreasury);
+        
+        // To Receiver by transferring ETH
+        (bool success, ) = receiver.call{value: ethReceived, gas: 100_000}("");
```

Comment:
given its just an ETH transfer, it is fine to not specify a gas limit here like we do in the LiquidityPool Contract:
```
function _sendFund(address _recipient, uint256 _amount) internal {
        uint256 balanace = address(this).balance;
        (bool sent, ) = _recipient.call{value: _amount}("");
        require(sent && address(this).balance == balanace - _amount, "SendFail");
    }
```
```suggestion
        (bool success, ) = receiver.call{value: ethReceived}("");

Reply:
If we have a reason to keep it, we could lower it to 50k and still be safe for basic operations in the receiver fallback

Reply:
thanks. reduced to 10k